src/C-interface/bml_utilities.c

```diff
@@ -529,3 +529,18 @@ bml_write_bml_matrix(
             break;
     }
 }
+
+// compute largest int "i" such that i*i <= x
+int
+bml_sqrtint(
+    const int x)
+{
+    int i = 1;
+    int result = 1;
+    while (result < x)
+    {
+        i++;
+        result = i * i;
+    }
+    return i;
+}
```

Comment:
```suggestion
{
    int i = 1;
    int result = 4;
    while (result < x)
    {
        i++;
        result = (i + 1) * (i + 1);
    }
    return i;
}
```

Reply:
@nicolasbock Not sure why you are suggesting that... Seems more complicated to me

Reply:
Sorry, I should have added a comment to explain :smile: The function as is does not work. For example `sqrtint(10) -> 4`.

Reply:
I see. I guess the comment doesn't match the code, even though it works for all the square int (4, 9, 16,...) which is what we really care about

Reply:
Thanks! Yes, this makes more sense now :smile: